XGBoost is a boosting library with parallel, GPU, and distributed execution support. It has help many machin learning engineers and data scientists to win Kaggle competitions. 

Furthermore, it provides an interface that resembles scikit-learn interface. Thus, someone already farmilar with the interface is able to quickly utilize the library. Additionally, it allows for very fine control over the ensemble's creation. It supports monotonic contraints (that is the predicted value should only increase or decrease, relative to a specific feature), as well as feature interaction constraints (for example, if a decision tree creates a node that splits by age, it should not use sex as a splitting feature for all children of that specific node). Finally it adds an additional regularization parameter gamma, which further reduces the overfitting capabilities of the gernerated ensemble.

In [1]:
!pip install xgboost
# Step 1: Import libraries and data
from sklearn.datasets import load_diabetes
from xgboost import XGBRegressor
from sklearn import metrics
import numpy as np
diabetes = load_diabetes()
train_size = 400
train_x, train_y = diabetes.data[:train_size], diabetes.target[:train_size]
test_x, test_y = diabetes.data[train_size:], diabetes.target[train_size:]
np.random.seed(123456)

In [2]:
#Step 2: Create the ensemble
ensemble_size = 200
ensemble = XGBRegressor(n_estimators=ensemble_size, n_jobs=4, max_depth=1, learning_rate=0.1, objective='reg:squarederror')

In [3]:
# Step 3: Evaluate the ensemble
ensemble.fit(train_x,train_y)
predictions = ensemble.predict(test_x)

In [4]:
#Step 4: print the metrics
r2= metrics.r2_score(test_y, predictions)
mse = metrics.mean_squared_error(test_y,predictions)
print('XGBoost: ')
print('R-squared: %.2f' % r2)
print('MSE: %.2f'%mse)

XGBoost: 
R-squared: 0.65
MSE: 1932.91


XGBoost achives an R-squared of 0.65 and an MSE of 1932.9. This is the best performance out of all the boosting method we did test and implement.